src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/Designer/DataTransferTypes.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.VisualStudio.Imaging.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug.Designer
+{
+    internal sealed record PropertyValueChangeRequest(
+        string PropertyName,
+        object Value);
+
+    internal sealed record ProfileType(
+        string CommandName,
+        string DisplayName,
+        string? HelpUrl,
+        ImageMoniker Icon);
+
+    /// <summary>
+    /// Represents a launch profile and all associated properties and values.
+    /// </summary>
+    /// <remarks>
+    /// The design of this type (as well as ILaunchProfileEditorClientSession"
+    /// assumes that the debugger command name associated with a Profile never changes.
+    /// The set of properties associated with a profile depends on the debugger command;
+    /// if the command can change than the set of properties on a profile can change,
+    /// not just their values as in the current design. If we adopt an alternative
+    /// design that allows changing the command we will also need to update ILaunchProfileClientEditorSession"
+    /// to support adding/removing properties.
+    /// </remarks>
+    /// <param name="Name">
+    /// The unique name of the profile.
+    /// </param>
```

Comment:
Nice! I was wondering how to document record parameters. Seems obvious in retrospect 😃

Reply:
This is why I had to update the compiler package to a 3.9.0 version--this isn't supported in 3.8.0.